lib/Migration/OwncloudGuestsMigration.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * SPDX-FileCopyrightText: 2025 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+namespace OCA\Guests\Migration;
+
+use OCA\Guests\UserBackend;
+use OCP\IConfig;
+use OCP\IUserManager;
+use OCP\Migration\IOutput;
+use OCP\Migration\IRepairStep;
+use Psr\Log\LoggerInterface;
+
+class OwncloudGuestsMigration implements IRepairStep {
+
+	protected $cachedUserIDs = [];
+
+	public function __construct(
+		protected LoggerInterface $logger,
+		protected IConfig $config,
+		protected IUserManager $userManager,
+		protected UserBackend $guestUserBackend,
+	) {
+	}
+
+	public function getName(): string {
+		return 'Converts owncloud guests to nextcloud guests';
+	}
+
+	public function run(IOutput $output): void {
+		if (!$this->shouldRun()) {
+			$this->logger->info('Skipping owncloud guests migration, no owncloud guests found');
+			return;
+		} else {
+			$this->runStep($output);
+		}
+	}
+
+	protected function shouldRun(): bool {
+		$this->cachedUserIDs = $this->config->getUsersForUserValue('owncloud', 'isGuest', '1');
+		$count = count($this->cachedUserIDs);
+		if ($count === 0) {
+			return false;
+		}
+
+		$this->logger->info("Found $count owncloud guests to migrate");
+
+		return true;
+	}
+
+	/**
+	 * For each old guest user, we do the following:
+	 * 1. Create a new guest with the same username and password
+	 * 2. register a new transfer
```

Comment:
What do you mean by "register a new transfer"?

Reply:
There is a transfer service to handle files ownership change

Reply:
I don't see it in the code, am I missing something?

Reply:
This is leftover comment from before things were implemented.

Since the uid stays the same no data needs to be migrated

Reply:
> I don't see it in the code, am I missing something?

That was my oriignal plan, I didn't see it wasn't used in the end :see_no_evil: 
My bad